PR: https://github.com/odoo/odoo/pull/233595

From: cdb34988e2afdcf7bd81d79b7b3ab9f8ffbb1702
From: dhrs-odoo
Date: 2025-12-03 12:53:42

Breaking data model changes scores: del:2 + add:1, change matches:
-    salary_proposed = fields.Float("Proposed", aggregator="avg", help="Salary Proposed by the Organisation", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
-    salary_expected = fields.Float("Expected", aggregator="avg", help="Salary Expected by Applicant", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
+    tag_ids = fields.Many2many('hr.job.tag', string='Tags')

Total Changes: 51

[IMP] hr_recruitment: add job tags and improve salary fields

* Changed field types of `salary_expected` and `salary_proposed` from Float to
  Monetary for better currency handling.
* Added new selection field `payment_interval` to define salary payment.
* Created new model `hr.job.tags` to manage job-related tags.
* Added Many2many field `tag_ids` on `hr.job` to link job positions with tags.
* Updated list and kanban views of Job Positions to display associated tags.
* Added new window action "Job Tags" under Job Positions menu.

Task ID: 5212975

closes odoo/odoo#233595

Related: odoo/enterprise#99861
Related: odoo/upgrade#8906
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/__init__.py
+++ b/addons/hr_recruitment/models/__init__.py
@@ -22,3 +22,4 @@ from . import ir_attachment
 from . import ir_ui_menu
 from . import mail_activity_plan
 from . import hr_job_platform
+from . import hr_job_tag

--- a/addons/hr_recruitment/models/hr_applicant.py
+++ b/addons/hr_recruitment/models/hr_applicant.py
@@ -83,6 +83,7 @@ class HrApplicant(models.Model):
     last_stage_id = fields.Many2one('hr.recruitment.stage', "Last Stage",
                                     help="Stage of the applicant before being in the current stage. Used for lost cases analysis.")
     categ_ids = fields.Many2many('hr.applicant.category', string="Tags")
+    currency_id = fields.Many2one('res.currency', string='Currency', related='company_id.currency_id')
     company_id = fields.Many2one('res.company', "Company", compute='_compute_company', store=True, readonly=False, tracking=True)
     user_id = fields.Many2one(
         'res.users', "Recruiter", compute='_compute_user', domain="[('share', '=', False), ('company_ids', 'in', company_id)]",
@@ -92,10 +93,18 @@ class HrApplicant(models.Model):
     date_last_stage_update = fields.Datetime("Last Stage Update", index=True, default=fields.Datetime.now)
     priority = fields.Selection(AVAILABLE_PRIORITIES, "Evaluation", default='0')
     job_id = fields.Many2one('hr.job', "Job Position", domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", tracking=True, index=True, copy=False)
+    salary_proposed = fields.Monetary("Proposed", aggregator="avg", currency_field='currency_id', help="Salary Proposed by the Organisation", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
     salary_proposed_extra = fields.Char("Proposed Salary Extra", help="Salary Proposed by the Organisation, extra advantages", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
+    salary_expected = fields.Monetary("Expected", aggregator="avg", currency_field='currency_id', help="Salary Expected by Applicant", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
     salary_expected_extra = fields.Char("Expected Salary Extra", help="Salary Expected by Applicant, extra advantages", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
-    salary_proposed = fields.Float("Proposed", aggregator="avg", help="Salary Proposed by the Organisation", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
-    salary_expected = fields.Float("Expected", aggregator="avg", help="Salary Expected by Applicant", tracking=True, groups="hr_recruitment.group_hr_recruitment_user")
+    schedule_pay = fields.Selection([
+        ('hourly', 'Hour'),
+        ('daily', 'Day'),
+        ('weekly', 'Week'),
+        ('biweekly', 'Bi-Week'),
+        ('monthly', 'Month'),
+        ('yearly', 'Year'),
+    ], string='Schedule Pay', default='monthly', required=True, groups="hr_recruitment.group_hr_recruitment_user")
     department_id = fields.Many2one(
         'hr.department', "Department", compute='_compute_department', store=True, readonly=False,
         domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", tracking=True)
@@ -640,6 +649,9 @@ class HrApplicant(models.Model):
                 vals['date_open'] = fields.Datetime.now()
             if vals.get('email_from'):
                 vals['email_from'] = vals['email_from'].strip()
+            if vals.get('job_id'):
+                job = self.env['hr.job'].browse(vals['job_id'])
+                vals['schedule_pay'] = job.payment_interval or 'monthly'
         applicants = super().create(vals_list)
         applicants.sudo().interviewer_ids._create_recruitment_interviewers()
 

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -84,6 +84,19 @@ class HrJob(models.Model):
         store=True)
 
     job_source_ids = fields.One2many('hr.recruitment.source', 'job_id', groups="hr_recruitment.group_hr_recruitment_interviewer")
+    tag_ids = fields.Many2many('hr.job.tag', string='Tags')
+    currency_id = fields.Many2one(
+        'res.currency', related='company_id.currency_id', readonly=True, groups="hr.group_hr_user,hr_recruitment.group_hr_recruitment_user")
+    salary_min = fields.Monetary('Minimum Salary', currency_field='currency_id', groups="hr.group_hr_user,hr_recruitment.group_hr_recruitment_user")
+    salary_max = fields.Monetary('Maximum Salary', currency_field='currency_id', groups="hr.group_hr_user,hr_recruitment.group_hr_recruitment_user")
+    payment_interval = fields.Selection([
+        ('hourly', 'Hour'),
+        ('daily', 'Day'),
+        ('weekly', 'Week'),
+        ('biweekly', 'Bi-Week'),
+        ('monthly', 'Month'),
+        ('yearly', 'Year'),
+    ], string='Scheduled Pay', default='monthly', required=True, groups="hr.group_hr_user,hr_recruitment.group_hr_recruitment_user")
 
     @api.depends('application_ids.date_closed')
     def _compute_no_of_hired_employee(self):
@@ -289,6 +302,11 @@ class HrJob(models.Model):
             })
         return values
 
+    @api.onchange('salary_min', 'salary_max')
+    def _onchange_salary(self):
+        if self.salary_min > self.salary_max:
+            self.salary_min, self.salary_max = self.salary_max, self.salary_min
+
     @api.model_create_multi
     def create(self, vals_list):
         for vals in vals_list:

--- a/None
+++ b/addons/hr_recruitment/models/hr_job_tag.py
@@ -0,0 +1,16 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from random import randint
+
+from odoo import fields, models
+
+
+class HrJobTag(models.Model):
+    _name = 'hr.job.tag'
+    _description = "Tag"
+
+    def _get_default_color(self):
+        return randint(1, 11)
+
+    name = fields.Char("Tag Name", required=True)
+    color = fields.Integer(string='Color Index', default=_get_default_color)
